docs/hello_nextflow/01_hello_world.md

```diff
@@ -0,0 +1,448 @@
+-   **`.command.run`**: Full script run by Nextflow to execute the process call
+-   **`.command.sh`**: The command that was run by the process call call
+-   **`.exitcode`**: The exit code resulting from the command
+
+The `.command.sh` file is especially useful because it tells you what command Nextflow actually executed.
+In this case it's very straightforward, but later in the course you'll see commands that involve some interpolation of variables.
+When you're dealing with that, you need to be able to check exactly what was run, especially when troubleshooting an issue.
+
+The actual output of the `sayHello` process is `output.txt`.
+Open it and you will find the `Hello World!` greeting, which was the expected result of our minimalist workflow.
+
+### Takeaway
+
+You know how to decipher a simple Nextflow script, run it and find the output and relevant log files in the work directory.
+
+### What's next?
+
+Learn to 'publish' outputs to a location outside the work directory for convenience.
+
+---
+
+## 3. Publish outputs
+
+As you have just learned, the output produced by our pipeline is buried in a working directory several layers deep.
+This is done on purpose; Nextflow is in control of this directory and we are not supposed to interact with it.
+
+But that makes it inconvenient to retrieve outputs that we care about.
+
+Fortunately, Nextflow provides a way to manage this more conveniently, called the `publishDir` directive, which acts at the process level.
+This directive tells Nextflow to copy the output(s) of the process to a designated output directory.
+It allows us to retrieve the desired output file without having to drill down into the work directory.
+
+### 3.1. Add a `publishDir` directive to the `sayHello` process
+
+In the workflow script file `hello-world.nf`, make the following code modification:
+
+_Before:_
+
+```groovy title="hello-world.nf" linenums="6"
+process sayHello {
+
+    output:
+        path 'output.txt'
+```
+
+_After:_
+
+```groovy title="hello-world.nf" linenums="6"
+process sayHello {
+
+    publishDir 'results', mode: 'copy'
+
+    output:
+        path 'output.txt'
+```
+
+### 3.2. Run the workflow again
+
+Now run the modified workflow script:
+
+```bash
+nextflow run hello-world.nf
+```
+
+The log output should look very familiar:
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+ ┃ Launching `hello-world.nf` [mighty_lovelace] DSL2 - revision: 6654bc1327
+
+executor >  local (1)
+[10/15498d] sayHello [100%] 1 of 1 ✔
+```
+
+This time, Nextflow has created a new directory called `results/`.
+Our `output.txt` file is in this directory.
+If you check the contents it should match the output in the work subdirectory.
+This is how we move results files outside of the working directories conveniently.
+
+It is also possible to set the `publishDir` directive to make a symbolic link to the file instead of actually copying it.
+This is preferable when you're dealing with very large files.
```

Comment:
```suggestion
This is preferable when you're dealing with very large files you don't need to retain longer term.
```

I still maintain that the practical applications of the default symliking are few.